dataprofiler/profilers/profile_builder.py

```diff
@@ -535,3 +536,14 @@ def _update_profile_from_chunk(df, profile=None, sample_size=None,
                 )
 
         return profile
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+
+    def __enter__(self):
+        return self
+
+    def close(self):
+        config.data_labeler_loaded = False
+        config.existing_data_labeler = None
```

Comment:
I don't believe this should be associated with the profiles as the existence of the labeler should be independent of profiles.